scripts/xdcc_autoget.pl

```diff
@@ -0,0 +1,711 @@
+		{
+			if ($fileLine eq $tempLine)	#if entry in file and arguments
+			{
+				$hTmp{$fileLine}++;	#set flag to not copy
+			}
+			print TEMP2 qq{$fileLine} unless $hTmp{$fileLine};	#copy other lines to other temp file
+		}
+		close(TEMP);
+	}
+	close(TEMP2);
+	copy("/tmp/temp2", $file);	#rewrite old file
+	copy($file, "/tmp/temp");
+	unlink "$file";
+	open(TEMP, "<", "/tmp/temp");
+	open(SEARCHES, ">", $file);
+	%hTmp = ();
+	while (my $sLine = <TEMP>)		#remove duplicate lines
+	{
+		next if $sLine =~ m/^\s*$/;  #remove empty lines. Without this, still destroys empty lines except for the first one.
+		$sLine=~s/^\s+//;            #strip leading/trailing whitespace
+		$sLine=~s/\s+$//;
+		print FILE qq{$sLine\n} unless ($hTmp{$sLine}++);
+	}
+	unlink "/tmp/temp";
+	unlink "/tmp/temp2";
+	close(FILE);
+}
+
+sub ag_add	#add search terms
+{
+	&ag_server;
+	my @args = quotewords('\s+', 0, $_[0]);	#split arguments (words in brackets not seperated)
+	if ($#args < 0)
+	{
+		Irssi::print "AG | too few arguments";
+		Irssi::print "AG | usage: ag_add <search terms>";
+		return;
+	}
+	&ag_parseadd($searchesfilename, @args);
+}
+
+sub ag_rem	#remove ssearch terms
+{
+	&ag_server;
+	my @args = quotewords('\s+', 0, $_[0]);
+	if ($#args < 0)
+	{
+		Irssi::print "AG | too few arguments";
+		Irssi::print "AG | usage: ag_rem <search terms>";
+		return;
+	}
+	&ag_parserem($searchesfilename, @args);
+}
+
+sub ag_botadd	#add bots
+{
+	&ag_server;
+	my @args = quotewords('\s+', 0, $_[0]);
+	if ($#args < 0)
+	{
+		Irssi::print "AG | too few arguments";
+		Irssi::print "AG | usage: ag_botsadd <bots>";
+		return;
+	}
+	&ag_parseadd($botsfilename, @args);
+}
+
+sub ag_botrem	#remove bots
+{
+	&ag_server;
+	my @args = quotewords('\s+', 0, $_[0]);
+	if ($#args < 0)
+	{
+		Irssi::print "AG | too few arguments";
+		Irssi::print "AG | usage: ag_rem <search terms>";
+		return;
+	}
+	&ag_parserem($botsfilename, @args);
+}
+
+sub ag_run	#main loop
+{
+	if($runningflag == 0)
+	{
+		$runningflag = 1;
+		&ag_getbots;
+		&ag_getterms;
+		if($#bots < 0 or $#terms < 0) {Irssi::print "AG | No bots or no search terms added. Halting"; &ag_stop;}
+		else 
+		{
+			Irssi::print "AG | Search and get cycle Initiated";
+			&ag_search;
+		}
+	}
+	else {Irssi::print "AG | Another Instance is already running";}
+}
+
+sub ag_stop
+{
+	Irssi::signal_remove("dcc get receive", "ag_opendcc");
+	Irssi::signal_remove("message irc notice", "ag_getmsg");
+
+	foreach my $to (@totags)
+	{
+		Irssi::timeout_remove($to);
+	}
+	@totags = ();
+
+	&ag_message("msg $bots[$botcounter] $cancelprefix");
+
+	if($runningflag == 1)
+	{
+		$runningflag = 0;
+		Irssi::print "AG | Killed";
+	}
+	$msgflag = 1;
+	$termisepisodicflag = 0;
+	$formatflag = 1;
+	$reqpackflag = 0;
+	$downloadflag = 0;
+	$newpackflag = 1;
+	$dccflag = 0;
+	@terms = ();
+	@bots = ();
+	@packs = ();
+	@finished = ();
+	$termcounter = 0;
+	$botcounter = 0;	
+	$packcounter = 0;
+	$episode = 1;
+}
+
+sub ag_reset
```

Comment:
fyi, this is just github being silly and not removing that old comment although you had updated the code